ostd/src/trap/softirq.rs

```diff
@@ -95,12 +95,12 @@ impl SoftIrqLine {
 /// A slice that stores the [`SoftIrqLine`]s, whose ID is equal to its offset in the slice.
 static LINES: Once<[SoftIrqLine; SoftIrqLine::NR_LINES as usize]> = Once::new();
 
-/// Initializes the softirq lines.
-///
-/// # Safety
-///
-/// This function must be called only once.
-pub unsafe fn init() {
+/// Initializes the software IRQ lines if not initialized.
+pub fn init() {
+    if LINES.get().is_some() {
```

Comment:
What is the motivation of this change?

Reply:
It's not necessarily unsafe.

Reply:
It has been removed, since it's not related to this PR.

Reply:
I suppose the `unsafe` keyword can be removed. But you do not have to modify the body and guard against calling `init` multiple times (with `if LINES.get().is_some()`). It is a logic error to call `init` more than once. In this case, `Once` will simply panic.